scripts/can_bundle.py

```diff
@@ -68,15 +68,17 @@ def fetch_bundle(source: str, bundle_uuid: str, bundle_version: str) -> Bundle:
     for catalog in config.catalogs:
         plugin = plugin_for(catalog)
         sources = set(map(str, plugin.sources))
-        if source in sources:
-            source = plugin.resolve_source(source)
-            fqid = SourcedBundleFQID(source=source,
-                                     uuid=bundle_uuid,
-                                     version=bundle_version)
-            bundle = plugin.fetch_bundle(fqid)
-            logger.info('Fetched bundle %r version %r from catalog %r.',
-                        fqid.uuid, fqid.version, catalog)
-            return bundle
+        parsed_source = plugin.resolve_source(source)
```

Comment:
Having the `plugin.resolve_source` moved before the `if source in sources` means an error will occur if the source is not in valid format for the catalog.

This can occur when a `canned` repository source is used, and `AZUL_CATALOGS` defines the `tdr` repository catalogs before the `canned` repository catalog. In this case a `tdr` repository  plugin will attempt `plugin.resolve_source` with the `canned` repository source and fail
```
Traceback (most recent call last):
...
  File "/Users/daniel/repo/azul3/src/azul/terra.py", line 151, in parse
    service, project, name, prefix = spec.split(':')
```

A work-around is to make sure the `canned` repository catalog is defined first in `AZUL_CATALOGS`, however this requirement is undesirable hack.

See attached file [canned_repository.txt](https://github.com/DataBiosphere/azul/files/7227321/canned_repository.txt) for `canned` repository catalog configuration and console log while using `can_bundle.py`.

Reply:
Alright, I think i got it this time, i used the catalog config within the txt, but i did not move the canned catalog above the normal catalogs.

```bash
export_environment.py: Setting AZUL_CATALOGS to '{"dcp2": {"atlas": "hca", "internal": false, "plugins": {"metadata": {"name": "hca"}, "repository": {"name": "tdr"}}}, "dcp2ebi": {"atlas": "hca", "internal": false, "plugins": {"metadata": {"name": "hca"}, "repository": {"name": "tdr"}}}, "lungmap": {"atlas": "lungmap", "internal": false, "plugins": {"metadata": {"name": "hca"}, "repository": {"name": "tdr"}}}, "it2": {"atlas": "hca", "internal": true, "plugins": {"metadata": {"name": "hca"}, "repository": {"name": "tdr"}}}, "it2ebi": {"atlas": "hca", "internal": true, "plugins": {"metadata": {"name": "hca"}, "repository": {"name": "tdr"}}}, "it3lungmap": {"atlas": "lungmap", "internal": true, "plugins": {"metadata": {"name": "hca"}, "repository": {"name": "tdr"}}}, "canned": {"atlas": "hca", "internal": false, "plugins": {"metadata": {"name": "hca"}, "repository": {"name": "canned"}}}}'
export_environment.py: Setting azul_canned_sources to https://github.com/HumanCellAtlas/schema-test-data/tree/55628953e4b3a24a7d7798569b6082032bd07a6b/tests:
```